### PR TITLE
[Metrics] Add metrics for scheduling

### PR DESCRIPTION
**Pull request by zhongchun**
## What do these changes do?
* Add scheduling, band and op metrics
```
    mars.scheduling.submitted_subtask_count: The count of submitted subtasks to all bands.
    mars.scheduling.finished_subtask_count: The count of finished subtasks of all bands.
    mars.scheduling.canceled_subtask_count: The count of canceled subtasks of all bands.
    mars.band.submitted_subtask_number: The number of submitted subtask to a band.
    mars.band.unsubmitted_subtask_number: The number of unsubmitted subtask to a band.
    mars.operand.executed_number: The number of executed operands.
```
* Optimize metrics arguments check

## Related issue number

Fixes #xxxx

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

**Comment by qinxuye** [comment]
Some metrics named with suffix `count`, but some with `number`, looks a bit weird though, is it possible to unify the naming?

**Comment by zhongchun** [comment]
> Some metrics named with suffix `count`, but some with `number`, looks a bit weird though, is it possible to unify the naming?

Thanks for you suggestion.
`_count` means metric type is `Counter`, `_number` means its type is `Gauge`. Just to distinguish them from their types.

**Comment by qinxuye** [comment]
> > Some metrics named with suffix `count`, but some with `number`, looks a bit weird though, is it possible to unify the naming?
> 
> Thanks for you suggestion. `_count` means metric type is `Counter`, `_number` means its type is `Gauge`. Just to distinguish them from their types.

OK, can we have a document about the naming specification about metrics? Just add it to https://github.com/mars-project/mars/tree/master/docs/source/development ?

**Review by qinxuye — COMMENTED** [review]
LGTM overall, like I said, we can add a document about adding metrics.

**Comment by zhongchun** [comment]
> LGTM overall, like I said, we can add a document about adding metrics.

Ok, I'll add docs soon.

**Comment by zhongchun** [comment]
> > > Some metrics named with suffix `count`, but some with `number`, looks a bit weird though, is it possible to unify the naming?
> > 
> > 
> > Thanks for you suggestion. `_count` means metric type is `Counter`, `_number` means its type is `Gauge`. Just to distinguish them from their types.
> 
> OK, can we have a document about the naming specification about metrics? Just add it to https://github.com/mars-project/mars/tree/master/docs/source/development ?

Added in #2955, please take a look.

File: mars/core/operand/core.py [hunk 1]
```diff
@@ -39,6 +40,11 @@
 _op_type_to_size_estimator: Dict[Type[OperandType], Callable] = dict()
 
 
+op_executed_number = Metrics.counter(
+    "mars.operand.executed_number", "The number of executed operands.", ("op",)
```
Comment (wjsi) [inline mars/core/operand/core.py:44]: The number of *what* kind of objects executed? Please make the name of the counter and the variable more explicit.

**Review by qinxuye — APPROVED** [review]
LGTM

File: mars/core/operand/core.py (hunk 1 above)
Comment (zhongchun) [reply mars/core/operand/core.py:44]: How about `mars.operator_executed_number`? And the description is `The number of executed operators.`

Comment (wjsi) [reply mars/core/operand/core.py:44]: executed_operator_number

**Review by wjsi — APPROVED** [review]
LGTM

File: mars/core/operand/core.py (hunk 1 above)
Comment (wjsi) [reply mars/core/operand/core.py:44]: Sorry but I've ignored the prefix.. Already merged.

**Pull request merged**